pkg/clusterlogging/clusterlogging.go

```diff
@@ -0,0 +1,170 @@
+package clusterlogging
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	clov1 "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Builder provides struct for clusterLogging object.
+type Builder struct {
+	// ClusterLogging definition. Used to create clusterLogging object with minimum set of required elements.
+	Definition *clov1.ClusterLogging
+	// Created clusterLogging object on the cluster.
+	Object *clov1.ClusterLogging
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before clusterLogging object is created.
+	errorMsg string
+}
+
+// NewBuilder method creates new instance of builder.
+func NewBuilder(
+	apiClient *clients.Settings, name, nsname string) *Builder {
+	glog.V(100).Infof("Initializing new clusterLogging structure with the following params: name: %s, namespace: %s",
+		name, nsname)
+
+	builder := &Builder{
+		apiClient: apiClient,
+		Definition: &clov1.ClusterLogging{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the clusterLogging is empty")
+
+		builder.errorMsg = "The clusterLogging 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the clusterLogging is empty")
+
+		builder.errorMsg = "The clusterLogging 'namespace' cannot be empty"
+	}
+
+	return builder
+}
+
```

Comment:
Don't you need Pull function which will pull existing object to Builder? Refer to example: https://github.com/openshift-kni/eco-goinfra/blob/d37bdd48d168b4737f647c4144bfff5aced29c17/pkg/pod/pod.go#L92C2-L92C2

Reply:
I have Get instead, it's just the same

Reply:
No get is the different. In order to use get you need to create Builder **first** and then request Get  however Pull will create builder and load object automatically.